### ISSUE TITLE
test-proxy config locate should emit supported path

### ISSUE THREAD

**Issue by heaths** [Azure/azure-sdk-tools#7240]
When using `test-proxy config locate -a assets.json`, on Windows it emits an invalid path that not every program will be able to use e.g.,

```
C:/src/azure-sdk-for-net/.assets/fRNnAo5UpV/net
```

See broken scenario: 

<img width="596" alt="image" src="https://github.com/Azure/azure-sdk-tools/assets/1532486/49e24730-c782-4479-8713-de0a868d3cfa">

`test-proxy` should emit appropriate platform paths at least when displaying information to the user.

**Comment by scbedd** [comment]
Ok I'm missing something. I _must_ be missing something.

```bash
semick@CPC-scbed-QJJWS:~/azure-sdk-for-python$ test-proxy config locate -a sdk/tables/azure-data-tables
Running proxy version is Azure.Sdk.Tools.TestProxy 1.0.0-dev.20231012.1
git --version
/home/semick/azure-sdk-for-python/.assets/v5O3LEDhWR/python
semick@CPC-scbed-QJJWS:~/azure-sdk-for-python$ cd /home/semick/azure-sdk-for-python/.assets/v5O3LEDhWR/python
semick@CPC-scbed-QJJWS:~/azure-sdk-for-python/.assets/v5O3LEDhWR/python$
```

```powershell
(venv) C:/repo/azure-sdk-for-python [feature/add-optional-env]|>test-proxy config locate -a sdk/tables/azure-data-tables
Running proxy version is Azure.Sdk.Tools.TestProxy 20231013.1
git --version
C:/repo/azure-sdk-for-python/.assets/v5O3LEDhWR/python
```

**Comment by heaths** [comment]
Yes, in my case the problem was something probably having to do with worktrees. In my case, it showed a folder path on my primary git repo (the one where `.git` is a folder), but the `.assets` directory was actually created in a worktree (where `.git` is a file).

That said, not every shell e.g., `cmd.exe`, will automatically translate `/` to `\`:

<img width="315" alt="image" src="https://github.com/Azure/azure-sdk-tools/assets/1532486/baf78d6f-2ca9-4387-9e61-61e827a218f0">

Any paths emitted to the console should use the OS directory char e.g., `System.IO.Path.DirectorySeparatorChar`.

**Comment by scbedd** [comment]
> Any paths emitted to the console should use the OS directory char e.g., System.IO.Path.DirectorySeparatorChar.

I thought it would be safer to map everything to `/`. It's actually a TIL that it doesn't work in `cmd` in all cases. I just went and tried on `cmd` and you're absolutely right.


```cmd
REM pushd fails with this pattern
pushd sdk/core/azure-core
REM cd works with this pattern
cd sdk/core/azure-core
```

I'll make an update to `NormalizedString` that just replaces with system directory separator char. That's a bit annoyhing.

**Comment by heaths** [comment]
It also would matter for anything you might pass it to that doesn't normalize separate chars for cross-plat support. I was going to show an example using `notepad.exe`, but apparently the new one does support `/`. The old version didn't.

**Comment by scbedd** [comment]
> In my case, it showed a folder path on my primary git repo (the one where .git is a folder), but the .assets directory was actually created in a worktree (where .git is a file).

When your `.git` is a file, it should still map the assets to `.assets` folder at the root of the repo. We just identify the folder that we call "root" by the presence of a `.git` folder or a `.git` file.

> but the .assets directory was actually created in a worktree

This part is a bit confusing to me. Can you describe what you mean by "created in the worktree"?

**Comment by heaths** [comment]
Consider this:

1. C:\src\azure-sdk-for-net - primary repo that did not have Key Vault .NET SDK test assets yet
2. C:\src\azure-sdk-for-net2 - worktree that did have KV test assets

When I ran `test-proxy config locate -a sdk/keyvault/Azure.Security.KeyVault.Keys/assets.json` in the first directory, it showed me a path relative to the first directory but only actuall existed in the second directory. That was my initial confusion as to why I was getting the "folder doesn't exist" error but, yes, PS itself does canonicalize the path (which I knew but wasn't thinking about). It was indeed a red herring for *this* problem, but a problem itself.

Can I safely just delete the .assets directory? I can clean up and try to get a fresh repo to open a new issue with as much detail as I can collect.

**Comment by scbedd** [comment]
> Can I safely just delete the .assets directory? I can clean up and try to get a fresh repo to open a new issue with as much detail as I can collect.

Definitely. Not if you have pending uncommitted stuff of course, but aside from that it should figure out and re-restore if necessary.

**Comment by scbedd** [comment]
> When I ran test-proxy config locate -a sdk/keyvault/Azure.Security.KeyVault.Keys/assets.json in the first directory, it showed me a path relative to the first directory but only actually existed in the second directory.

That is CRAZY and I am definitely happy to fix it if you're seeing that. Based on the code walk through for `AscendToRepoRoot`, I'm a bit confused how it happened, but it obviously happened _somehow!_

Please file that issue I'll definitely fix it. RE: this one I'll have a PR out next week.

**Comment by heaths** [comment]
I'll try to repro later. Maybe my assumption is off and there were other things going on. But I do tend to work in each worktree (primary or otherwise) in different terminal tabs, so I'm positive at least on the result - just not how it happened.
